### PR TITLE
Use language.php instead of dummy (only english) language

### LINKED ISSUE #94: Installing from git shows only the English language
When running the hypha.php installer, a list of languages is presented to the user to select the default language. This language list is based on the `languages.php` file contained *inside* `hypha.php`. When running from a downloaded bundle, this file is complete. However, when running from git, the file is already on the filesystem, and there is only a [dummy version](https://github.com/PlanBCode/hypha/blob/124a05cb7ab1756c128c52edb5a11803588b223b/hypha.php#L501) inside hypha.php, to at least not break things.

Ideally, this would simply read the file from the filesystem when it is not present inside `hypha.php`, rather than needing a dummy file, so the complete list of languages is always available. When both versions are available, I think the one from `hypha.php` should take precedence, to allow overwriting an existing install.

### PR DESCRIPTION
**Pull request by RoukePouw**
Fixes #94

Place languages in separate json data file

### CODE REVIEW

File: hypha.php [hunk 1]
```diff
@@ -85,7 +85,7 @@ function index($dir = '.') {
 		if ($handle = opendir($dir)) {
 			while (false !== ($file = readdir($handle))) {
 				if ($file != "." && $file != "..") {
-					if (isAllowedFile($dir.'/'.$file)) $index[ltrim($dir.'/'.$file, "./")] = filemtime($dir.'/'.$file);
+					if (isAllowedFile($dir.'/'.$file)) $index[preg_replace("/^\.\//i","",$dir.'/'.$file)] = filemtime($dir.'/'.$file);
```
Comment (matthijskooijman) [inline hypha.php]: I think this change is unrelated to the rest of the commit (IIRC it fixes installation of the .htaccess file?). Would be better to have this in a separate commit (separate PR is not needed).

File: hypha.php [hunk 2]
```diff
@@ -293,10 +293,8 @@ function install() {
 			if (!is_writable('.')) $errorMessage.= 'php has no write access to the hypa installation directory on the server<br/>';
 
 			// extract list of languages
-			$iso639 = data('system/core/language.php');
-			$p1 = strpos($iso639, "json_decode('");
-			$p2 = strpos($iso639, "', true);", $p1);
-			$iso639 = json_decode(preg_replace('/\\\\\'/', '\'', substr($iso639, $p1+13, $p2-$p1-13)));
+			$iso639 = data('system/languages/languages.json');
+			$iso639 = json_decode(preg_replace('/\\\\\'/', '\'', $iso639));
```
Comment (matthijskooijman) [inline hypha.php]: Why do we still need this replacement? Isn't the content of `languages.json` valid json as-is?

File: hypha.php [hunk 3]
```diff
@@ -420,7 +418,9 @@ function data($name) {
 		switch ($name) {
 			//START_OF_DATA
 			case 'index': return '';
-			case 'system/core/language.php': return "json_decode('{\"en\": \"English\"}', true);";
+			case 'system/core/language.php':
+        require_once('system/core/language.php');
+        return $isoLangList;
```
Comment (matthijskooijman) [inline hypha.php]: Shouldn't this have a case for `languages.json` rather than `language.php` now? And that could just return the `file_get_contents()` for the existing `languages.json`?

File: system/core/language.php [hunk 4]
```diff
@@ -33,4 +33,9 @@ function languageOptionList($select, $omit) {
...
+  
+  if(file_exists('system/languages/languages.json')){
+    $isoLangList = json_decode(file_get_contents('system/languages/languages.json'), true);
+  }else{
```
Comment (matthijskooijman) [inline system/core/language.php]: Do we need this check and else? AFAICS we always have a `languages.json` whenever `languages.php` also exists, so we could just remove the if?

File: hypha.php (hunk 2 above)
Comment (matthijskooijman) [reply hypha.php]: Ah, it seems these `\'` is present in the `languages.json` file you added. These were probably needed when the json was contained in a PHP `''` string, but this is no longer relevant. I would rather remove these backslashes from the `languages.json` file and remove the replacement above.

File: hypha.php (hunk 1 above)
Comment (matthijskooijman) [reply hypha.php]: This commit could probably be set to close #92.

Comment (matthijskooijman) [reply hypha.php]: This comment is still unresolved.

File: hypha.php (hunk 2 above)
Comment (matthijskooijman) [reply hypha.php]: This is resolved.

File: system/core/language.php (hunk 4 above)
Comment (matthijskooijman) [reply system/core/language.php]: This is resolved.

File: hypha.php (hunk 3 above)
Comment (matthijskooijman) [reply hypha.php]: This is partly resolved, but I'll add a new comment for what's left.

File: hypha.php [hunk 5]
```diff
@@ -420,7 +418,7 @@ function data($name) {
 		switch ($name) {
 			//START_OF_DATA
 			case 'index': return '';
-			case 'system/core/language.php': return "json_decode('{\"en\": \"English\"}', true);";
+			case 'system/languages/languages.json': return json_decode(file_get_contents('system/languages/languages.json'), true);
```
Comment (matthijskooijman) [inline hypha.php]: This now returns the decoded json (e.g. a PHP array), whereas the version that is generated when packing up hypha has this section replaced with actual file data, so that changes the return value. IOW, when you pack up hypha, the code no longer works.

To fix that, this should just return the result of `file_get_contents()`, and the `json_decode()` should be moved to the caller of `data()` elsewhere. Might be good to also test this packing up (though I'm not 100% sure it is working in master).

Comment (laurensmartina) [reply hypha.php]: fixed

File: hypha.php (hunk 1 above)
Comment (laurensmartina) [reply hypha.php]: done
